crates/walrus-service/src/node.rs

```diff
@@ -2223,8 +2270,20 @@ impl StorageNodeInner {
         Ok(metadata)
     }
 
-    fn current_event_epoch(&self) -> Epoch {
-        self.latest_event_epoch.load(Ordering::SeqCst)
+    /// Returns the latest event epoch.
+    /// If the storage node hasn't processed the first event yet, this function will block until
+    /// the first event is processed, so that the current event epoch is set.
+    async fn current_event_epoch(&self) -> Result<Epoch, watch::error::RecvError> {
```

Comment:
Question: What could cause the `RecvError` here? Can this actually happen?

Reply:
RecvError is thrown when one side of the channel is closed. I think this theoretically can only happen when a node is turning down. However, I'd rather we populate the error instead of panicing the node.